lib/base/io-engine.hpp

```diff
@@ -137,22 +141,25 @@ class IoEngine
 	boost::asio::executor_work_guard<boost::asio::io_context::executor_type> m_KeepAlive;
 	std::vector<std::thread> m_Threads;
 	boost::asio::deadline_timer m_AlreadyExpiredTimer;
+
 	std::atomic_int_fast32_t m_CpuBoundSemaphore;
+	std::mutex m_CpuBoundWaitingMutex;
+	std::vector<std::pair<boost::asio::io_context::strand, Shared<AsioConditionVariable>::Ptr>> m_CpuBoundWaiting;
 };
 
 class TerminateIoThread : public std::exception
 {
 };
 
 /**
- * Condition variable which doesn't block I/O threads
+ * Python's threading.Event equivalent which doesn't block I/O threads
```

Comment:
Please summarize what it does without completely relying on the reference to Python. That should rather be an additional "inspired by threading.Event from Python" hint.